Sources/ArcGISToolkit/Components/FeatureTemplatePicker/FeatureTemplatePicker.swift

```diff
@@ -0,0 +1,334 @@
+                    )
+                }
+                sections.append(
+                    .init(table: table, infos: infos)
+                )
+            }
+            return sections.filter { !$0.infos.isEmpty }
+        }
+        
+        /// Re-filters based on the templates or the search text.
+        private func templatesOrSearchTextDidChange() {
+            featureTemplateSections = unfilteredFeatureTemplateSections.map {
+                $0.filtered(by: searchText)
+            }
+            .filter { !$0.infos.isEmpty }
+            showContentUnavailable = unfilteredFeatureTemplateSections.isEmpty
+            showNoTemplatesFound = !showContentUnavailable && featureTemplateSections.isEmpty
+        }
+    }
+}
+
+private extension Array<Layer> {
+    /// A flattened list of the nested layers that this array of layers may contain.
+    var flattened: [Layer] {
+        flatMap { layer in
+            guard let groupLayer = layer as? GroupLayer else { return [layer] }
+            return groupLayer.layers.flattened
+        }
+    }
+}
+
+private extension GeoModel {
+    /// All the layers in the geo model.
+    var layers: [Layer] {
+        operationalLayers
+        + (basemap?.baseLayers ?? [])
+        + (basemap?.referenceLayers ?? [])
+    }
+    
+    /// A flattened list of the layers in the geo model.
+    var flattenedLayers: [Layer] {
+        layers.flattened
+    }
+    
+    /// All the feature layers in the geo model.
+    var featureLayers: [FeatureLayer] {
+        flattenedLayers
+            .compactMap { $0 as? FeatureLayer }
+    }
+    
+    /// A list containing tuples of the feature layers and the associated
+    /// ArcGIS feature tables in the geo model.
+    var arcGISFeatureLayersAndTables: [(layer: FeatureLayer, table: ArcGISFeatureTable)] {
+        featureLayers
+            .map { (layer: $0, table: $0.featureTable) }
+            .compactMap { tuple in
+                if let table = tuple.table as? ArcGISFeatureTable {
+                    return (layer: tuple.layer, table: table)
+                } else {
+                    return nil
+                }
+            }
+    }
+}
+
+private extension ArcGISFeatureTable {
+    /// A flattened list of the feature templates in the table.
+    var allTemplates: [FeatureTemplate] {
+        let typeTemplates = featureTypes
+            .lazy
+            .flatMap { $0.templates }
+        return featureTemplates + typeTemplates
+    }
+}
+
+/// A value that represents a section in the feature template picker.
+private struct FeatureTemplateSectionInfo: Identifiable {
+    let table: ArcGISFeatureTable
+    var infos: [FeatureTemplateInfo]
+    let id: UUID = UUID()
+    
+    func filtered(by searchText: String) -> FeatureTemplateSectionInfo {
+        guard !searchText.isEmpty else { return self }
+        var copy = self
+        copy.infos = copy.infos.filter { $0.template.name.lowercased().contains(searchText.lowercased()) }
```

Comment:
Elsewhere in the the Toolkit we've used [`localizedStandardContains(_:)`](https://developer.apple.com/documentation/swift/stringprotocol/localizedstandardcontains(_:)) for search similar search functionalities.

```suggestion
        copy.infos = copy.infos.filter { $0.template.name.localizedStandardContains(searchText) }
```

> This is the most appropriate method for doing user-level string searches, similar to how searches are done generally in the system. The search is locale-aware, case and diacritic insensitive. The exact list of search options applied may change over time.